$OpenBSD$

OpenBSD patch for patch-src-Simplex-Messaging-Transport-KeepAlive_hs.

--- src/Simplex/Messaging/Transport/KeepAlive.hs.orig
+++ src/Simplex/Messaging/Transport/KeepAlive.hs
@@ -33,7 +33,11 @@
 _SOL_TCP :: CInt
 _SOL_TCP = 6
 
-#if defined(mingw32_HOST_OS)
+#if defined(openbsd_HOST_OS)
+setSocketKeepAlive :: Socket -> KeepAliveOpts -> IO ()
+setSocketKeepAlive sock _ = setSocketOption sock KeepAlive 1
+
+#elif defined(mingw32_HOST_OS)
 -- Windows
 
 -- The values are copied from windows::Win32::Networking::WinSock
@@ -63,11 +67,5 @@
 
 #endif
 
-setSocketKeepAlive :: Socket -> KeepAliveOpts -> IO ()
-setSocketKeepAlive sock KeepAliveOpts {keepCnt, keepIdle, keepIntvl} = do
-  setSocketOption sock KeepAlive 1
-  setSocketOption sock (SockOpt _SOL_TCP _TCP_KEEPIDLE) keepIdle
-  setSocketOption sock (SockOpt _SOL_TCP _TCP_KEEPINTVL) keepIntvl
-  setSocketOption sock (SockOpt _SOL_TCP _TCP_KEEPCNT) keepCnt
 
 $(J.deriveJSON defaultJSON ''KeepAliveOpts)
